dateparser/date_parser.py

```diff
@@ -32,10 +34,65 @@ def __add__(self, other):
 parser.relativedelta.relativedelta = new_relativedelta
 
 
+class new_parser(parser.parser):
+    """
+    Implements an alternate parse method which supports preference to dates in future and past.
+    For more see issue #36
+    """
+
+    def parse(self, timestr, default=None, ignoretz=False, prefer_future=None, **kwargs):
+        # timestr needs to be a buffer as required by _parse
+        timestr = timestr if not isinstance(timestr, str) else StringIO(timestr)
+
+        # Set right prefer_future var
+        prefer_future = prefer_future if prefer_future is not None \
+            else settings.PREFER_DATES_FROM_FUTURE
+
+        # Parse timestr
+        res = self._parse(timestr, **kwargs)
+
+        if res is None:
+            raise ValueError("unknown string format")
+
+        # Fill in missing date
+        given_fields = []  # fields set by
+        new_date = default
+
+        for field in ['year', 'month', 'day', 'hour', 'minute', 'second', 'microsecond']:
+            value = getattr(res, field)
+            if value is not None:
+                new_date = new_date.replace(**{field: value})
+                given_fields.append(field + 's')
+        if res.weekday and not res.day:
+            new_date = new_date + new_relativedelta(weekday=res.weekday)
+
+        # Correct date to if prefer in future of past
+        if prefer_future is not None:
+            for field in ['microseconds', 'seconds', 'minutes', 'hours', 'days',
+                          'weeks', 'months', 'years']:
+                # Can't override given field
+                if field in given_fields:
+                    continue
+
+                delta = relativedelta(**{field: 1})
+
+                if (prefer_future is False and new_date >= default and new_date - delta < default) or\
+                   (prefer_future is True and new_date <= default and new_date + delta > default):
+
+                    new_date = new_date - delta if prefer_future is False else new_date + delta
+                    break
+
+        # Clean hour and minutes, etc in case not defined
+        if not res.hour and not res.minute and not res.second and not res.microsecond:
+            new_date = new_date.replace(hour=0, minute=0, second=0, microsecond=0)
+
+        return new_date
+
+
 def dateutil_parse(date_string, **kwargs):
     """Wrapper function around dateutil.parser.parse
     """
-    today = datetime.utcnow().replace(hour=0, minute=0, second=0, microsecond=0)
+    today = datetime.utcnow()
```

Comment:
This is essential. This code sets the clock to midnite. However, what if default is 16:00 and I want the future 15:00, it would result the same day, however it should be the day before.